src/index.ts

```diff
@@ -105,6 +91,7 @@ export interface AppMetadata {
   name: string;
   version: string;
   args: string;
+  htsFile: HtsFileMetadata | undefined;
```

Comment:
how about `htsFile?: HtsFileMetadata;` and only add the key to the object if `HtsFileMetadata` is available?